### PR TITLE
Added redirect field (id and comment)

### PR DESCRIPTION
**Pull request by jcpitre**
**Summary:**
Closes https://github.com/MobilityData/mobility-feed-api/issues/149
Associated PR in mobility catalogs: https://github.com/MobilityData/mobility-database-catalogs/pull/309

- Added a redirect comment to the `redirectingid` table in the DB
- Modified populate script to take into account the `redirect.id` and `redirect.comment` columns in the csv.
- Modified the openapi specs to add a redirect id and comment to the schemas
- Modified the API code to return the data.

**Expected behavior:** 

With https://github.com/MobilityData/mobility-database-catalogs/pull/309 the csv should have redirect.id and redirect.comment columns

- The DB should now have a `redirect_comment` added to the `redirectingid` table.
- The populate script should be able to read the 2 new columns and fill the redirecting table properly.
- Calling the API for `feeds`, `gtfs_feeds` and `gtfs_rt_feeds` should now return a `redirects` array.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: api/src/scripts/populate_db.py [hunk 1]
```diff
@@ -232,6 +226,50 @@ def add_entity(entity, priority):
...
+                    continue
+                if comments:
+                    comment = comments.pop(0)
+                else:
+                    comment = ""
+
+                target_stable_id = f"mdb-{mdb_source_id}"
+                target_feed = feed_map.get(target_stable_id, None)
+
+                if target_feed:
```
Comment (davidgamez) [inline api/src/scripts/populate_db.py:263]: For safety, we should prevent a recursive redirection (feed.id == target_feed.id)

File: api/src/scripts/populate_db.py [hunk 2]
```diff
@@ -122,6 +123,10 @@ def add_entity(entity, priority):
 
         if self.df is None:
             return
+
+        # Keep a dict (map) of stable_id -> feed so we can reference the feeds when processing the static_reference
+        # and the redirects.
+        feed_map = {}
```
[the remark this answers is not in the document]
Comment (jcpitre) [reply api/src/scripts/populate_db.py:129]: We do. Later in the code we have:
```
referenced_feed = feed_map.get(static_reference_mdb_id, None)
           if referenced_feed:
                 referenced_feed.gtfs_rt_feeds.append(feed)
```
I could find other ways to optimize, but I am not sure it's worth the trouble

File: docs/DatabaseCatalogAPI.yaml [hunk 3]
```diff
@@ -233,6 +233,17 @@ paths:
 
 components:
   schemas:
+    Redirect:
+      type: object
+      properties:
+        target_id:
+          description: ID of the feed to which the current feed is being redirected.
+          type: string
+          example: mdb_10
```
[the remark this answers is not in the document]
Comment (jcpitre) [reply docs/DatabaseCatalogAPI.yaml]: Done

File: api/src/scripts/populate_db.py (hunk 1 above)
Comment (jcpitre) [reply api/src/scripts/populate_db.py:263]: Added a first level check (feed = redirect feed) but we should also check for indirect recursion (feed1 -> feed2 -> feed1)

**Review by davidgamez — APPROVED** [review]
LGTM!

File: docs/DatabaseCatalogAPI.yaml [hunk 4]
```diff
@@ -233,13 +233,24 @@ paths:
 
 components:
   schemas:
+    Redirect:
+      type: object
+      properties:
+        target_id:
+          description: ID of the feed to which the current feed is being redirected.
```
Comment (emmambd) [inline docs/DatabaseCatalogAPI.yaml:240]: Minor edit of description for clarity: `The feed ID that should be used in replacement of the current one.`

File: docs/DatabaseCatalogAPI_IAP.yaml [hunk 5]
```diff
@@ -233,13 +233,24 @@ paths:
 
 components:
   schemas:
+    Redirect:
+      type: object
+      properties:
+        target_id:
+          description: ID of the feed to which the current feed is being redirected.
```
Comment (emmambd) [inline docs/DatabaseCatalogAPI_IAP.yaml:240]: Minor edit of description for clarity: `The feed ID that should be used in replacement of the current one.`

**Review by emmambd — COMMENTED** [review]
@jcpitre 1 suggested change to the description in the API schema for clarity - otherwise looks great!

**Pull request merged**